includes/modules/core/command/ModeCommand.hpp

```diff
@@ -0,0 +1,138 @@
+#ifndef FT_IRC_CORE_MODE_COMMAND
+#define FT_IRC_CORE_MODE_COMMAND
+
+#include "ft_irc.hpp"
+
+#include "core/Irc.hpp"
+
+#include "api/ResponseTypes.hpp"
+#include "api/User.hpp"
+#include "api/command/CommandExecutor.hpp"
+
+#include "api/ResponseTypes.hpp"
+
+class ModeCommand : public CommandExecutor {
+
+	void execute(const Command& cmd, CommandSender& sender) {
+		User& user = static_cast<User&>(sender);
+
+		Channel& channel = cmd.getArg<Channel>("channel");
+		string *mode = cmd.getArg<string *>("mode");
+		vector<string *> *modeparams = cmd.getArg<vector<string *> *>("modeparams");
+
+		int sign = 0;
+		size_t param_i = 0;
+		Irc& server = Irc::getInstance();
+		flag &flags = channel.getFlag();
+
+		if (mode && !channel.isChanop(&user)) {
+			sender.send(ResponseTypes::ERR_CHANOPRIVSNEEDED(channel.getName().c_str()));
+			return ;
+		} else if (!channel.isOnChan(&user)) {
+			sender.send(ResponseTypes::ERR_USERNOTINCHANNEL(user.getName().c_str(), channel.getName().c_str()));
+			return ;
+		} else if (!mode) {
+			string mod;
+			string params;
+
+			if (flags.anonymous) {
+				mod += "a";
+			} if (flags.invite) {
+				mod += "i";
+			} if (flags.moderate) {
+				mod += "m";
+			} if (flags.outside_message) {
+				mod += "n";
+			} if (flags.quiet) {
+				mod += "q";
+			} if (flags.priv) {
+				mod += "p";
+			} if (flags.secret) {
+				mod += "s";
+			} if (flags.reop) {
+				mod += "r";
+			} if (flags.topic) {
+				mod += "t";
+			} if (!channel.getPassword().empty()) {
+				mod += "i";
+				params += channel.getPassword() + " ";
+			} if (flags.user_limit) {
+				mod += "i";
+				params += std::to_string(flags.user_limit);
+			}
+			sender.send(ResponseTypes::RPL_CHANNELMODEIS(channel.getName().c_str(), mod.c_str(), params.c_str()));
+			return ;
+		}
+
+		if ((*mode)[0] == '+') {
+			sign = 1;
+		} else if ((*mode)[0] == '-') {
+			sign = 2;
+		}
+
+		for (size_t i = sign ? 1 : 0; i < mode->size(); i++) {
+			if ((*mode)[i] == 'o') {
```

Comment:
switch is faster (it uses a binary tree)

Don't use a switch when there is less than 5 cases, because it will be slower than else if